Yoast/yoast-components
# SEO title input field should be single line

## Summary

This PR can be summarized in the following changelog entry:

* Implemented [draft-js-single-line-plugin](https://github.com/icelab/draft-js-single-line-plugin) in order to keep the Draft-js Editor from creating new blocks on enter presses. This results in a single-line editor. This now affects both the `SEO title` input field, and the `Meta description` input field.

## Relevant technical choices:

* Used [draft-js-single-line-plugin](https://github.com/icelab/draft-js-single-line-plugin) because preventing the default for Enter keydowns did not work.

## Test instructions

This PR can be tested by following these steps:

* Run `yarn install` and start yoast components.
* Type a sentence in the input field, and make it very long. The editor should remain one line high.
* Try to press `Enter` in the input fields, this should not create new lines.
* Try copy pasting text with linebreaks, `<br />`, etc, into the input fields. Never should you be able to create a new line.
* **NOTE**: there was an issue with the mention plugin entitities being stripped, which was fixed by passing an option to the single-line plugin. So please test that the auto-completing for replacevars still works.

Fixes https://github.com/Yoast/wordpress-seo/issues/9842
